javascript/javascript3/week3/screenshot.js

```diff
@@ -0,0 +1,197 @@
+class Screenshots {
+    constructor() { 
+        this.urlInput = document.querySelector('.postInput');   
+        this.deleteInput= document.querySelector('.deleteInput');   
+        this.inputs = document.querySelectorAll('input');   
+        this.currentList = document.querySelector('.currentList');  
+        this.fullCollection = document.querySelector('.fullCollection');
+
+        this.screenShotOptions = {
+            method: 'GET',
+            headers: {
+                'X-RapidAPI-Key': 'c4c71f72f4msh1b9d8f48e635255p185dabjsn218387ea6564',
+                'X-RapidAPI-Host': 'website-screenshot6.p.rapidapi.com'
+            }
+        };
+        
+        this.getListOptions = {
+            method: 'GET',	
+        };
+        
+        this.deleteListOptions = {
+            method: 'DELETE',	
+        };
+    }
+
+    renderScreen(id, screenshotUrl, url, user) {
+        const listElement = document.createElement('li');
+        listElement.classList.add('currentListItem');
+        listElement.innerHTML = `
+        ScreenshotUrl - <a href="${screenshotUrl}" target="_blank">${screenshotUrl}</a><br>
+        User URL - <a href="${url}" target="_blank">${url}</a><br>
+        Post id - ${id}<br>
+        User - ${user}
+        `;
+        //------------------------------------
+        this.currentList.append(listElement);
+
+        const img = document.createElement('img');
+        img.src = screenshotUrl;
+        listElement.appendChild(img);
+    }
+
+    renderCollection(id, screenshotUrl, url, user) {
+        const listElement = document.createElement('li');
+        listElement.classList.add('fullCollectionItem');
+        listElement.innerHTML = `id: ${id},<br> screenshotUrl: <a href="${screenshotUrl}" target="_blank">${screenshotUrl}</a>,<br> url: <a href="${url}" target="_blank">${url}</a>,<br> user: ${user}`;
+        this.fullCollection.append(listElement);
+    }
+
+    async getResource(url, options) {
+        let response = await fetch(url, options);
+        if (options.method === 'DELETE') {
+            return null;
```

Comment:
You might still want to handle errors if the method is DELETE - when we make a request of CRUD type it is nice to let the user know if the outcome of the operation was succesfull or not. 
It is a good idea to always wrap an `await` statement in a `try/catch` block. Now with fetch  there is a little caveat as you need to do additional error handling an d check by `response.ok`, like you did on line 55 (good job!). But here, checking by the method is is to early to exit this function and return null (btw, DELETE method can also return something :)